src/shogun/io/Serializable.h

```diff
@@ -145,10 +146,15 @@ template<class T> class StringListSerializable: public SGObject
 protected:
 	virtual void init()
 	{
-		set_generic<T>();
-		m_value = SGStringList<T>();
-		m_parameters->add_vector(&m_value.strings, &m_value.num_strings, "value",
-				"Serialized value");
+		if constexpr (type_internal::is_sg_primitive<T>::value)
+			set_generic<T>();
+		// FIXME(?): std::vector<bool> is not a container
+		// doesn't play well with the parameter framework
+		if constexpr (!std::is_same<T, bool>::value)
+			watch_param("value", &m_value);
+		else
+			SG_WARNING("std::vector<bool> is not currently serializable. "
+					   "Use SGVector<bool> instead.");
```

Comment:
@vigsterkr I copied this from the develop branch but tests still fail for SGStringList in SGObjectAll, so I am assuming there is more to this?